apps/client/src/app/export-github-panel/export-github-panel.component.ts

```diff
@@ -105,28 +105,31 @@ export class ExportGithubPanelComponent implements OnInit {
     });
 
     this.allFiles.sort((a, b) => (a.path > b.path) ? 1 : (a.path === b.path) ? 0 : -1);
+    this.allFiles.forEach((file: FileModel) => {
+      const mainFilePath = file.path.substring(0, file.path.lastIndexOf("."));
+      const foundFiles: FileModel[] = this.allFiles.filter(f => f.path.indexOf(mainFilePath) >= 0);
+
+      if(foundFiles.length > 1){
+        foundFiles.forEach((foundFile: FileModel) => {
+          if((this.responseFormat.indexOf("xml") >= 0 && foundFile.path.indexOf("json") >= 0) ||
```

Comment:
... think it needs to be more complex. this, for example, would delete `package-list.json` if the user selected "xml".
I think the way to solve this would be to look for duplicate file names (the same file name with both .json and .xml extension), and then delete the one that the represents the format the user _doesn't_ have selected. Here's some pseudo logic to get you started:
```
const duplicateFiles = this.allFiles.filter(f => {
  const fileNameWithoutExtension = ...
  return this.allFiles.filter(next => {
    const nextFileNameWithoutExtension = ...
    return nextFileNameWithoutExtension === fileNameWithoutExtension;
  }).length > 1;
});

duplicateFiles.forEach(f => {
  const fileNameExtension = ...
  
  if (this.responseFormat.indexOf('.json') > 0 && fileNameExtension.toLowerCase() != 'json') {
    // flag file as deleted
  } else if (this.responseFormat.indexOf('.xml') > 0 && fileNameExtension.toLowerCase() != 'xml') {
    // flag file as deleted
  }
});
```